pkg/vaultclient/vaultclient.go

```diff
@@ -57,6 +57,34 @@ func (v *VaultClient) ListKV(path string) ([]string, error) {
 	return keyResponse.Keys, nil
 }
 
+func (v *VaultClient) ListKVRecursively(path string) ([]string, error) {
+	paths := []string{path}
+	var result []string
+	for _, path := range paths {
+		children, err := v.ListKV(path)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list %s: %w", path, err)
+		}
+		for _, child := range children {
+			if strings.HasSuffix(child, "/") {
+				// staticcheck complains `this result of append is never used, except maybe in other appends` but
+				// we do use it in the initial range.
+				// nolint: staticcheck
+				paths = append(paths, child)
```

Comment:
is it valid to change the slice you're looping over?

Reply:
I wondered the same, hence the test. Seems to be working fine